core/src/main/java/com/kin/ecosystem/core/data/blockchain/BlockchainSourceImpl.java

```diff
@@ -101,24 +108,81 @@ public static BlockchainSourceImpl getInstance() {
 	}
 
 	@Override
-	public void createAccount() throws BlockchainException {
-		createKinAccountIfNeeded();
+	public void loadAccount(String kinUserId) throws BlockchainException {
+		migrateToMultipleUsers(kinUserId);
+		createOrLoadAccount(kinUserId);
 		initBalance();
 	}
 
-	private void createKinAccountIfNeeded() throws BlockchainException {
-		if (kinClient.hasAccount()) {
+	/**
+	 * Support backward compatibility,
+	 * load the old account and migrate to current multiple users implementation.
+	 */
+	private void migrateToMultipleUsers(String kinUserId) {
+		final int accountIndex = local.getAccountIndex();
```

Comment:
is there a risk of migration from older SDK versions before backup and restore that didn't have the concept of accountIndex?

Reply:
we need to start thinking about standardize migration and keep track of migration history so we can make sure that we also support non-incremental updates

Reply:
there is no risk.

Reply:
👍